languages/bash.scm

```diff
@@ -328,3 +340,45 @@
 
   (#delimiter! ";")
 )
+
+;; Function Definitions
+
+; NOTE Much of the formatting work for function definitions is done by
+; whatever already-defined queries apply to the function body (e.g.,
+; (compound_statement), etc.). All we do here is ensure functions get
+; their own line and put a space between its name and the body.
+(function_definition) @prepend_hardline
+
+(function_definition
+  body: _ @prepend_space
+)
+
+; NOTE The "function" keyword in function definitions is optional and
+; thus usually considered redundant. Therefore we delete it, if it's
+; present in the input.
+(function_definition
+  .
+  "function" @delete
+)
+
+;; Variable Declaration, Assignment and Expansion
+
+; Declaration and assignment on a new line.
+(declaration_command) @prepend_hardline
+(variable_assignment) @prepend_empty_softline
+
+; Multiple variables can be exported (and assigned) at once
+(declaration_command
+  .
+  "export"
+  [(variable_name) (variable_assignment)] @prepend_space
+)
+
+; NOTE The (simple_expansion), for `$foo`, and (expansion), for `${foo}`
+; and friends, node types exist. We consider them as leaves (see above).
+; However, it would be _really_ nice if we could write a query that
+; converts all (simple_expansions) into (expansions). It can almost be
+; done with delimiters, but it doesn't quite work :( For example:
+;
+; (simple_expansion (variable_name) @prepend_delimiter (#delimiter! "{"))
+; (simple_expansion (variable_name) @append_delimiter (#delimiter! "}"))
```

Comment:
Why doesn't it?

Reply:
A `(simple_expansion)` is made up of `$` and `(variable_name)`. Adding the closing `}` works fine, but I cannot write a query that interposes the `{` between the sigil and name.

The above, commented out query, renders `{$foo}`. I tried being really specific, with anchors, like:

```scm
(simple_expansion
  .
  "$" @append_delimiter
  .
  (variable_name) ; OR: @prepend_delimiter
  .

  (#delimiter! "{")
)
```

...but that gives `$foo}{` and `{$foo}`, respectively.

Reply:
Hmm, that is odd. Does it make a difference if you put `"$"` in parens?

Reply:
No luck, I'm afraid

Reply:
🤷